pkg/rbd/rbd_util.go

```diff
@@ -185,12 +186,17 @@ func rbdManagerTaskDeleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.C
 			strings.Contains(string(output), rbdTaskRemoveCmdInvalidString2) {
 			klog.Infof(util.Log(ctx, "cluster with cluster ID (%s) does not support Ceph manager based rbd image"+
 				" deletion (minimum ceph version required is v14.2.3)"), pOpts.ClusterID)
-			return false, err
 		} else if strings.HasPrefix(string(output), rbdTaskRemoveCmdAccessDeniedMessage) {
 			klog.Infof(util.Log(ctx, "access denied to Ceph MGR-based RBD image deletion "+
 				"on cluster ID (%s)"), pOpts.ClusterID)
-			return false, err
+		} else if strings.HasPrefix(string(output), rbdTaskRemoveCmdPermissionDeniedMessage) {
```

Comment:
how about converting the output to a string if an error occurs? because in each if statement we are converting the output to string and doing the comparison. and also as if-else is increasing can we convert this to switch based check